docs/mp/cors/cors.adoc

```diff
@@ -0,0 +1,250 @@
+
+== API
+
+=== The `@CrossOrigin` Annotation
+
+Adding CORS behavior to your Helidon MP application involves just a few simple steps.
+
+For reach resource class in your application:
+
+. Identify the resources and subresources--in other words, the paths--declared in the resource class which you want to support CORS.
+. For each of those resources and subresources which should support CORS:
+.. Find or create a Java method annotated with `@OPTIONS` and with the correct `@Path`.
+.. To that `@OPTIONS` Java method add a Helidon link:{mp-cors-javadoc-base-url}/io/helidon/microprofile/cors/CrossOrigin.html}[`@CrossOrigin`] annotation that describes the cross-origin sharing you want for that resource.
+
+[NOTE]
+.Using @CrossOrigin Correctly
+====
+Use the `@CrossOrigin` annotation _only_ on methods which also have the `@OPTIONS` annotation. Remember that the `@CrossOrigin` settings apply to a given path and therefore to all Java resource methods which share that path.
+
+Helidon MP aborts the server start-up if you use the `@CrossOrigin` annotation on a resource method other than an `@OPTIONS` method.
+
+For an informal look at the reasons for applying the `@CrossOrigin` annotation to the `@OPTIONS` method, instead of another
+method, see xref:{rootdir}/mp/cors/why-options.adoc[Why `@OPTIONS`?].
+====
+
+The xref:config-table[configuration table] below describes the attributes of the `@CrossOrigin` annotation.
+
+
+== Configuration
+
+You can define CORS behavior--and you or your users can override behavior declared in your code--using configuration.
+
+For each resource you want to configure, add a section to `META-INF/microprofile-config.properties` file:
+
+.General form of CORS configuration
+[source,properties,subs="verbatim,quotes"]
+----
+cors.enabled= # <1>
+#
+# <2>
+cors.paths._i_.path-pattern= # <3>
+cors.paths._i_.allow-headers=
+cors.paths._i_.max-age-seconds=
+cors.paths._i_.allow-credentials=
+cors.paths._i_.allow-origins=
+cors.paths._i_.expose-headers=
+cors.paths._i_.allow-methods=
+cors.paths._i_.enabled= # <4>
+----
+<1> You can disable CORS processing for all resources by setting `cors.enabled` to `false`. Defaults to `true`.
+<2> Add a block for each resource you want to configure.
+The index `_i_` is an integer (0, 1, 2, etc.).
+<3> Specify the settings as needed to define the CORS behavior you want for that resource.
+<4> The `enabled` setting lets you control whether the system uses that set of CORS configuration.
+
+The system uses the index `_i_`, not the position in the config file, to identify the settings for a particular resource.
+
+Path patterns can be any expression accepted by the link:{webserver-javadoc-base-url}/io/helidon/webserver/PathMatcher.html[`PathMatcher`] class.
+
+NOTE: Be sure to assign index values to the entries in the order that you want Helidon to check them.
+Helidon CORS support searches the cross-origin entries in index order them until it finds an entry that
```

Comment:
Not sure what this means? 
Helidon CORS support searches the cross-origin entries in index order them until ...